libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -100,175 +364,1547 @@ namespace webassembly {
+          *
+          * @ingroup authorization
+          * @param account - name of the account to be tested.
+          *
+          * @retval true if the action has an auth with the account name.
+          * @retval false otherwise.
+         */
+         bool has_auth(account_name account) const;
+
+         /**
+          * Add the specified account to set of accounts to be notified.
+          *
+          * @ingroup authorization
+          * @param recipient - account to be notified.
+         */
+         void require_recipient(account_name recipient);
+
+         /**
+          * Verifies that n is an existing account.
+          *
+          * @ingroup authorization
+          * @param account - name of the account to check.
+          *
+          * @return true if the account exists.
+          * @return false otherwise.
+         */
+         bool is_account(account_name account) const;
+
+         /**
+          * Returns the time in microseconds from 1970 of the current block.
+          *
+          * @ingroup system
+          *
+          * @return time in microseconds from 1970 of the current block.
+         */
          uint64_t current_time() const;
+
+         /**
+          * Returns the transaction's publication time.
+          *
+          * @ingroup system
+          *
+          * @return time in microseconds from 1970 of the publication_time.
+         */
          uint64_t publication_time() const;
-         bool is_feature_activated(legacy_ptr<const digest_type>) const;
+
+         /**
+          * Check if specified protocol feature has been activated.
+          *
+          * @ingroup system
+          * @param feature_digest - digest of the protocol feature.
+          *
+          * @retval true if the specified protocol feature has been activated.
+          * @retval false otherwise.
+         */
+         bool is_feature_activated(legacy_ptr<const digest_type> feature_digest) const;
+
+         /**
+          * Return the name of the account that sent the current inline action.
+          *
+          * @ingroup system
+          * @return name of account that sent the current inline action (empty name if not called from inline action).
+         */
          name get_sender() const;
 
-         // context-free system api
+         /**
+          * Aborts processing of this action and unwinds all pending changes if the test condition is true.
+          *
+          * @ingroup context-free
+         */
          void abort() const;
-         void eosio_assert(bool, null_terminated_ptr) const;
-         void eosio_assert_message(bool, legacy_span<const char>) const;
-         void eosio_assert_code(bool, uint64_t) const;
-         void eosio_exit(int32_t) const;
 
-         // action api
-         int32_t read_action_data(legacy_span<char>) const;
+         /**
+          * Aborts processing of this action if the test condition is true.
```

Comment:
Is this correct?

Reply:
there is an exception being thrown 

`EOS_THROW( eosio_assert_message_exception, "assertion failure with message: ${s}", ("s",message) );`

but correct me if this is not the case or it requires a different wording

Reply:
I think the description is negated? Isn't the action aborted when the condition is _false_

Reply:
that's right, thanks for catching that.